lib/search.rb

```diff
@@ -840,11 +840,25 @@ def category_search
   def user_search
     return if SiteSetting.hide_user_profiles_from_public && !@guardian.user
 
-    users = User.includes(:user_search_data)
+    users = User
+      .includes(:user_search_data)
       .references(:user_search_data)
+
+    users = if UserField.where(searchable: true).present?
+      users
```

Comment:
This is adding a fair amount of work to every search - at least one extra query plus a join.

Suggestion: Add the custom field to the `user_search_data` via `SearchIndexer`, then you get this free?